demos/demo_trifinger_impedance_control.py

```diff
@@ -0,0 +1,106 @@
+import math
+import time
+import pinocchio
+import numpy as np
+
+from robot_wrapper import Robot
+
+import blmc_robots
+import robot_interfaces
+
+
+# Initialize pinocchio model
+robot = Robot(visualizer='gepetto',
+                  viscous_friction=0.0)
+robot.initViewer(loadModel=True)
+
+robot_model = robot.model
+robot_data = robot_model.createData()
+
+# Initialize real robot
+finger = blmc_robots.Robot(robot_interfaces.trifinger,
+                           blmc_robots.create_trifinger_backend,
+                           "trifinger.yml")
+finger.initialize()
+
+# End effector Ids
+finger_tip_links = ["finger_tip_link_0", "finger_tip_link_120", "finger_tip_link_240"]
+finger_tip_ids = [robot_model.getFrameId(name) for name in finger_tip_links]
+
+
+def next_target(target_type, i=None, t=None, idx=None):
+	# defining step targets
+	if target_type == "step":
+		z = [0.05, 0.2]
+		if t % 1001 == 0:
+			idx = (idx + 1) % 2
+		return z[idx], idx
+
+	# defining continuous, more smoother trajectory
+	if target_type == "continuous":
+		return (1 + math.sin(math.radians(i))) * 0.12 + 0.06, idx
+
+
+def get_torques(J, x_obs, x_des, v_obs, v_des, Kp, Kd):
+	'''
+	Torque = J.T * [ kp*(x-x_des) + kd*(x`-x`_des) ]
```

Comment:
This could have some more information than just some formula (what is it about? what is the meaning of arguments and return value?).